addons/account/models/account_move.py

```diff
@@ -990,14 +990,16 @@ def _recompute_dynamic_lines(self, recompute_all_taxes=False, recompute_tax_base
                                     or not depending of the field 'recompute_tax_line' in lines.
         '''
         for invoice in self:
+            recompute_invoice_tax_lines = recompute_all_taxes
             # Dispatch lines and pre-compute some aggregated values like taxes.
-            for line in invoice.line_ids:
-                if line.recompute_tax_line:
-                    recompute_all_taxes = True
-                    line.recompute_tax_line = False
+            if any(line.recompute_tax_line for line in invoice.line_ids) or\
+                invoice.mapped('line_ids.tax_ids').flatten_taxes_hierarchy() > invoice.mapped('line_ids.tax_line_id'):
```

Comment:
@epictete The key line is here, before this if anything change in the invoice line, some onchange methods set `recompute_tax_line` to True on one of the lines, but tax lines are not in invoice lines so no other onchage sets them on for recompute.